src/httpc.c

```diff
@@ -75,6 +75,27 @@ curl_easy_header_cb(char *buffer, size_t size, size_t nitems, void *ctx)
 {
 	struct httpc_request *req = (struct httpc_request *) ctx;
 	const size_t bytes = size * nitems;
+
+	/**
+	 * Skip saving response headers if redirect will be performed
+	 * (to get headers of the final response).
+	 *
+	 * From CURLOPT_HEADERFUNCTION man:
+	 * It's important to note that the callback will be invoked for the
+	 * headers of all responses received after initiating a request and
+	 * not just the final response. This includes all responses which
+	 * occur during authentication negotiation. If you need to operate
+	 * on only the headers from the final response, you will need to
+	 * collect headers in the callback yourself and use HTTP status
+	 * lines, for example, to delimit response boundaries.
+	 */
+	long response_code = 0;
+	curl_easy_getinfo(req->curl_request.easy, CURLINFO_RESPONSE_CODE, &response_code);
+	int is_redirected = (response_code / 100) == 3;
+	if (req->follow_location && is_redirected) {
+		return bytes;
+	}
```

Comment:
It should work since we don't limit [count of redirects](https://curl.se/libcurl/c/CURLOPT_MAXREDIRS.html).

(Also it is unclear, whether we'll add several status lines into the buffer — IOW, whether the response code is set before or after invoking this callback. I guess the status lines are skipped by the headers parser, so just noted.)

More general approach would be collecting of headers and discarding them if we meet the new status line. We'll not need the `req->follow_location` flag in this case and it'll work for limited count of redirects. OTOH, it'll lead to extra memcpy() calls.

I don't know, whether it worth to rework it. Just thoughts.